WordPress/gutenberg
# Duotone: Only output Duotone filters that are used

## What?
This is an attempt at fixing https://github.com/WordPress/gutenberg/issues/38299. The idea is that by outputting only the SVGs that are used, rather than all of the presets, we can remove a lot of unused SVGs on the frontend.

## Why?
Less code will make the page load faster.

## How?
1. Create a new WP_Gutenberg class.
2. Add an action to the WP_Theme_JSON class to make is possible to hook onto declarations.
3. Save a list of the CSS variables that duotone uses to reference SVGs, in a static array on the new class.
4. Add a filter to the WP_Theme_JSON class to make is possible to output SVGs from another class.
5. Output the SVGs that are saved in the static array created in step 1.

## Testing Instructions
1. Switch to a theme that has a default duotone applied to all blocks (you can use the Sherbet variation of TT3)
2. Add an image to a post
6. Confirm that the image has a duotone applied
7. Check the page source and check that none of the default duotone SVGs are being output.

### Testing Instructions for Keyboard
For step 3 above check that the source contains an SVG with an ID which is the same as the id that the CSS on an image references in its `filter` property.
For step 4 check that there are no other SVGs with filters in this format on the page:
`<filter id="wp-duotone-*">`
